temporalio/worker/_worker.py

```diff
@@ -690,7 +687,15 @@ async def run(self) -> None:
         also cancel the shutdown process. Therefore users are encouraged to use
         explicit shutdown instead.
         """
-        await self._plugin.run_worker(self)
+
+        def make_lambda(plugin, next):
+            return lambda w: plugin.run_worker(w, next)
+
+        next_function = lambda w: w._run()
+        for plugin in reversed(self.plugins):
+            next_function = make_lambda(plugin, next_function)
+
+        await next_function(self)
```

Comment:
Since this pattern gets repeated a few times, maybe should make a private helper in the plugin file?